libraries/AP_Volz_Protocol/AP_Volz_Protocol.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_Volz_Protocol.h"
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_Volz_Protocol::var_info[] = {
+
+    // @Param: CHAN_BITMASK
+    // @DisplayName: Channel Bitmask
+    // @Description: Enable of volz servo protocol to specific channels
+	// @Values: 0,1....
+    // @User: Standard
+    AP_GROUPINFO("BITMASK", 0, AP_Volz_Protocol, chan_bitmask,  0),
+
+    // @Param: MIN_PWM
+    // @DisplayName: Minimum PWM Mapping
+    // @Description: Enable of volz servo protocol to specific channels
+	// @Values: 0,1....
+    // @User: Standard
+    AP_GROUPINFO("MIN_PWM", 1, AP_Volz_Protocol, min_pwm,  900),
+
+    // @Param: MAX_PWM
+    // @DisplayName: Miximum PWM Mapping
+    // @Description: Enable of volz servo protocol to specific channels
+	// @Values: 0,1....
+    // @User: Standard
+    AP_GROUPINFO("MAX_PWM", 2, AP_Volz_Protocol, max_pwm,  2100),
+
+
+    AP_GROUPEND
+};
+
+// Constructor
+AP_Volz_Protocol::AP_Volz_Protocol()
+{
+	AP_Param::setup_object_defaults(this, var_info);
+}
+
+AP_Volz_Protocol AP_Volz_Protocol::create(){
+	return AP_Volz_Protocol{};
+}
+
+void AP_Volz_Protocol::init(const AP_SerialManager& serial_manager){
+	if(min_pwm < 900)
```

Comment:
these params can change during runtime so please move to update.

Reply:
did it as a safety, so you won't send a bad value during flight and might crush, it need a power cycle to get new values so you have to setup properly on ground.

Reply:
Redoing calibration without rebooting is an extremely common use case, and is worth supporting, the clamping to 900/2100 can be maintained in `AP_Volz_Protocol::update(void)` but given that it can change at anytime in flight precomputing all of this here does not offer any added value.

On a higher level though can I ask why it needs this clamping? You are using the min/max pwm to limit output range on all channels, while the same task can be accomplished by setting the exsisting SERVOx_MIN and SERVOx_MAX parameters. With this the servo channel library will already have managed all the min/max scaling for you, and you can safely proceed without worrying about it. (Keeping a clamp to 900/2100 would remain valid of course if Volz won't handle anything outside that range)